src/auth/auth-manager.ts

```diff
@@ -1,62 +1,110 @@
+  PublicClientApplication,
+} from '@azure/msal-browser';
 import { jwtDecode } from 'jwt-decode';
+import { msalConfig, SCOPE } from '../api/constants';
 import { TokenType } from '../enum';
-import {
-  AccessTokenDecodeType,
-  AuthenticationResponseType,
-  RequestUser,
-} from '../types';
-import { login, refreshSession } from '../api/api';
-
-export const logoutUser: Function = () => {
-  Object.keys(TokenType).forEach((token) =>
-    window.localStorage.removeItem(token)
-  );
-};
+import { tokenDecodeType, AuthenticationResponseType } from '../types';
 
-export const saveUserDataLocalStorage: Function = ({
-  id_token,
-  access_token,
-  refresh_token,
-}: AuthenticationResponseType) => {
-  if (refresh_token) {
-    window.localStorage.setItem(TokenType.refresh, refresh_token);
-  }
-  window.localStorage.setItem(TokenType.bearer, id_token);
-  window.localStorage.setItem(TokenType.access, access_token);
-};
+export const AuthManager = () => {
+  let msalInst: PublicClientApplication | null = null;
 
-export const getToken: Function = (typeToken: TokenType) => {
-  return window.localStorage.getItem(typeToken);
-};
+  const init: Function = async () => {
+    const msalInstance = new PublicClientApplication(msalConfig);
+    msalInst = msalInstance;
+    await msalInst.initialize();
+  };
 
-export const isValidAccessToken: Function = () => {
-  const token = getToken(TokenType.access);
-  if (!token) {
-    return false;
-  }
-  const accessTokenDecode: AccessTokenDecodeType = jwtDecode(token);
-  const expDate = new Date(accessTokenDecode?.exp * 1000);
-  return expDate > new Date();
-};
+  const currentAccount: Function = async () => {
+    const account: AccountInfo | undefined =
+      await msalInst?.getAllAccounts()[0];
+    return account;
+  };
+
+  const saveUserDataLocalStorage: Function = (
+    token: AuthenticationResponseType
+  ) => {
+    window.localStorage.setItem(TokenType.idToken, token?.idToken);
+    window.localStorage.setItem(TokenType.accessToken, token?.accessToken);
+  };
+
+  const getAccessToken: Function = () => {
+    return window.localStorage.getItem(TokenType.accessToken);
+  };
+
+  const getIdToken: Function = () => {
+    return window.localStorage.getItem(TokenType.idToken);
+  };
 
-export const refreshUserSession: Function = async () => {
-  try {
-    const refreshToken: string | null = getToken(TokenType.refresh);
-    if (refreshToken) {
-      const res: AuthenticationResponseType = await refreshSession(
-        refreshToken
-      );
-      saveUserDataLocalStorage({ ...res });
+  const loginUser: Function = async () => {
+    const result = await msalInst?.loginPopup();
+    saveUserDataLocalStorage(result);
+  };
+
+  const logoutUser: Function = async () => {
+    await msalInst?.logoutPopup({ account: await currentAccount() });
+
+    Object.keys(TokenType).forEach((token) =>
+      window.localStorage.removeItem(token)
+    );
+  };
+
+  const isValidToken: Function = () => {
```

Comment:
Here we have a generic name that tells us that we can use this method to validate any token. However, in the method itself we are checking only the idToken. So, if i want to use the method isValidToken for any other token, i cannot use it and i have to create another method say 'isValidAccessToken()'. Either rename the method or rewrite the function, so that it could check the validity of any token.